### PR TITLE
tpr-backed storage: conditionally strip ns when listing

### LINKED ISSUE #785: tpr-backed storage: list is unconditionally stripping namespace
This should be conditional, but it's not:

https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/storage/tpr/storage_interface.go#L417-L422

Unit tests aren't catching this because the unit tests are only using brokers-- which are not namespaced-- e.g. a case where this unconditional stripping of the namespace isn't a problem.

This is probably impetus to augment the unit tests for tpr-backed storage with tests similar to the ones that already exist, but for another type-- a namespaced type.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #785 

Also adds new unit tests to cover the gap that the bug in #785 slipped through

May need rebase if #788 gets merged first.

### CODE REVIEW

File: pkg/storage/tpr/storage_interface.go [hunk 1]
```diff
@@ -414,10 +414,12 @@ func (t *store) List(
...
 		return err
 	}
-	for i, obj := range objs {
-		if err := removeNamespace(obj); err != nil {
-			glog.Errorf("removing namespace from obj %d (%s)", i, err)
-			return err
+	if !t.hasNamespace {
+		for i, obj := range objs {
+			if err := removeNamespace(obj); err != nil {
+				glog.Errorf("removing namespace from obj %d (%s)", i, err)
```
Comment (duglin) [inline pkg/storage/tpr/storage_interface.go]: I know this was there before, but printing "i" doesn't seem helpful to someone trying to debug the log msg. Can we print the obj instead?

**Comment by duglin** [comment]
just one question, but LGTM to me either way

File: pkg/storage/tpr/storage_interface.go (hunk 1 above)
Comment (krancour) [inline pkg/storage/tpr/storage_interface.go]: I agree. And I would have happily fixed, but this has been merged already. Open a separate issue?

Comment (duglin) [inline pkg/storage/tpr/storage_interface.go]: sounds good to me - thanks

**Pull request merged**